src/Parallel/ArrayCollection/ReceiveDataForElement.hpp

```diff
@@ -38,14 +38,14 @@ struct ReceiveDataForElement {
                     const ElementId<Dim>& element_to_execute_on,
                     typename ReceiveTag::temporal_id instance,
                     ReceiveData receive_data) {
-    ERROR(
-        "The multi-node code hasn't been tested. It should work, but be aware "
-        "that I haven't tried yet.");
+    [[maybe_unused]] const size_t my_node = Parallel::my_node<size_t>(cache);
     auto& element_collection = db::get_mutable_reference<
         typename ParallelComponent::element_collection_tag>(
         make_not_null(&box));
-    // Note: We'll be able to do a counter-based check here too once that
-    // works for LTS in `SendDataToElement`
+
+    ASSERT(
+        element_collection.count(element_to_execute_on) == 1,
+        "ElementId " << element_to_execute_on << " is not on node " << my_node);
```

Comment:
[optional] You could put the `Parallel::my_node<size_t>(cache)` here to avoid the `[[maybe_unused]]` temporary